### PR TITLE
Implement glob builtins

### PR DESCRIPTION
**Pull request by ereslibre**
Fixes: https://github.com/kubewarden/policy-evaluator/issues/39

### CODE REVIEW

File: crates/burrego/src/opa/builtins/glob.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    serde_json::to_value(escape(input))
+        .map_err(|e| anyhow!("glob.quote_meta: cannot convert value into JSON: {:?}", e))
+}
+
+fn escape(s: &str) -> String {
+    let mut escaped = String::new();
+    for c in s.chars() {
+        match c {
+            '?' | '*' | '[' | ']' => {
```
Comment (flavio) [inline crates/burrego/src/opa/builtins/glob.rs:20]: Looking at the `glob.match` documentation I think we might have to handle also: `:`, `{` and `}`.

I haven't looked at the source code of opa though...

**Review by flavio — COMMENTED** [review]
I left one question.

As a final note, this doesn't implement `glob.match`, hence we cannot close https://github.com/kubewarden/policy-evaluator/issues/39

File: crates/burrego/src/opa/builtins/glob.rs (hunk 1 above)
Comment (ereslibre) [reply crates/burrego/src/opa/builtins/glob.rs:20]: The go implementation uses https://github.com/gobwas/glob. I'm going to expand the quote so we match all special characters.

Comment (ereslibre) [reply crates/burrego/src/opa/builtins/glob.rs:20]: Their implementation check the glob special chars and escapes by prepending a `\`: https://github.com/open-policy-agent/opa/blob/581ba9c456be551b584759ae91f99ae6ec6b347b/vendor/github.com/gobwas/glob/glob.go#L68-L77.

The list of special characters is determined by: https://github.com/open-policy-agent/opa/blob/581ba9c456be551b584759ae91f99ae6ec6b347b/vendor/github.com/gobwas/glob/syntax/lexer/lexer.go#L23-L35.

I'm just copy-catting this behavior here, updated the PR.

**Comment by ereslibre** [comment]
> As a final note, this doesn't implement glob.match, hence we cannot close #39

We can close it, this is about `glob` builtins. https://github.com/kubewarden/policy-evaluator/pull/57 was targeting `regex` builtins, and in that case, that issue had `References:` instead of `Fix:` to signal exactly that.

However, `glob.quote_meta` is the only builtin for the `glob` group, and we are good by closing #39 after this merges.

**Pull request merged**